apps/design/backend/src/app.ts

```diff
@@ -221,35 +221,22 @@ function buildApi({ store }: { store: Store }) {
       return streamToBuffer(pdf);
     },
 
-    async exportElectionPackage(input: {
-      electionId: Id;
-    }): Promise<{ zipContents: Buffer; electionHash: string }> {
-      const { election, layoutOptions, systemSettings } = store.getElection(
-        input.electionId
-      );
-      const { electionDefinition } = layOutAllBallotStyles({
-        election,
-        // Ballot type and ballot mode shouldn't change the election definition, so
-        // it doesn't matter what we pass here
-        ballotType: BallotType.Precinct,
-        ballotMode: 'test',
-        layoutOptions,
-      }).unsafeUnwrap();
+    getElectionPackage({ electionId }: { electionId: Id }): {
+      filePath?: string;
+      task?: BackgroundTask;
+    } {
+      return store.getElectionPackage(electionId);
+    },
 
-      const zip = new JsZip();
-      zip.file(
-        ElectionPackageFileName.ELECTION,
-        electionDefinition.electionData
-      );
-      zip.file(
-        ElectionPackageFileName.SYSTEM_SETTINGS,
-        JSON.stringify(systemSettings, null, 2)
+    exportElectionPackage({ electionId }: { electionId: Id }): void {
+      const electionPackageTaskId = store.createBackgroundTask(
+        'generate_election_package',
+        { electionId }
       );
```

Comment:
Should multiple requests with the same election ID generate multiple background tasks? Or do we want this endpoint to call something like `store.getOrCreateGenerateElectionPackageBackgroundTask(electionId)` that checks for an existing background task and sets it within a transaction so we don't have two requests where one of them wins and the other becomes a task without anyone using the result?

Reply:
Good call! Incorporated this into my actual PR: https://github.com/votingworks/vxsuite/pull/4462